Python/sysmodule.c

```diff
@@ -2861,6 +2861,62 @@ _PySys_InitCore(PyThreadState *tstate, PyObject *sysdict)
     return _PyStatus_ERR("can't initialize sys module");
 }
 
+#ifdef HAVE_FORK
+extern int _PyIO_buffered_at_fork_reinit(PyObject *);
+
+static int
+stdio_at_fork_reinit(_Py_Identifier *key)
+{
+
+    int ret = 0;
+
+    PyObject *isatty = NULL;
+    PyObject *buffer = NULL;
+
+    PyObject *stdio = _PySys_GetObjectId(key);
+
+    _Py_IDENTIFIER(isatty);
+    isatty = _PyObject_CallMethodIdNoArgs(stdio, &PyId_isatty);
+    if (isatty == NULL) {
+        PyErr_Clear();
+        goto end;
+    }
+    if (Py_IsFalse(isatty)) {
+        goto end;
+    }
+
+    _Py_IDENTIFIER(buffer);
+    if (_PyObject_LookupAttrId(stdio, &PyId_buffer, &buffer) < 0) {
+        /* stdout.buffer and stderr.buffer are not part of the
+         * TextIOBase API and may not exist in some implementations.
+         * If not present, no need to reinitialize their locks. */
```

Comment:
The function can raises an AttributeError, no? If yes, you must clear it: PyErr_Clear().

Reply:
Fixed! (The function is MUCH simpler now. Thanks for the tips.)